docs/puzzles/day16.md

```diff
@@ -1,11 +1,572 @@
 import Solver from "../../../../website/src/components/Solver.js"
 
-# Day 16
+# Day 16: Packet Decoder
+
+by [@tgodzik](https://github.com/tgodzik)
 
 ## Puzzle description
 
 https://adventofcode.com/2021/day/16
 
+## Part1: You've got mail!
+
+It seems that that we can split our problem into two parts. First, we need to
+parse the example into structures that we can later use to calculate out
+results.
+
+Let's start with defining data structures to use:
+
+```scala
+enum Packet(version: Int, typeId: Int):
+  case Literal(version: Int, exprs: Long) extends Packet(version, 4)
+  case Operator(version: Int, typeId: Int, exprs: List[Packet]) extends Packet(version, typeId)
+```
+
+`Packet.Literal` will represent simple literal packets, that contain only a
+value. We are using Long just in case based on the experience with previous
```

Comment:
just in case of what?

Reply:
Add more clarification `in case of large integers`